### PR TITLE
Fixed the bug with splice, it mutates the original array in desirable…

### PR DESCRIPTION
**Pull request by jallusparta**
… way. Also fixed the "required" prompt bug.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -149,7 +149,10 @@ class App extends React.Component {
             })
         } else {
             const pos = this.state.multiSelectArray.map(object => object.id).indexOf(choice.id)
-            const newMultiSelectArray = this.state.multiSelectArray.splice(pos, 1)
+            console.log(pos)
+            var newMultiSelectArray = this.state.multiSelectArray
```
Comment (VonLatvala) [inline src/App.jsx]: this should do the same thing as it did before this commit, you should create a `copy` of the array and then splice it, like this (also don't use var pls):
```const newMultiSelectArray = Object.assign({}, this.state.multiSelectArray).splice(pos, 1)```

Comment (jallusparta) [reply src/App.jsx]: That was exactly the problem earlier, as splice mutates the array in memory, and returns the piece of array that was removed. 

Earlier sets the removed part of the array as the new selection array:
const newMultiSelectArray = this.state.multiSelectArray.splice(pos, 1)

Comment (jallusparta) [reply src/App.jsx]: Splice returns the removed part of the array, and therefore creates the error we just had. If functional style is preferred, then another method should be used

Comment (jallusparta) [reply src/App.jsx]: Splice returns the removed part of the array, and mutates the original array in memory. Object assign just recreates the bug. Gotta find another method to change this to functional style.

Comment (jallusparta) [reply src/App.jsx]: Changed to filter

**Comment by VonLatvala** [comment]
`Array.filter`, good call!

**Pull request merged**